File: 0014-longest-common-prefix/0014-longest-common-prefix.js
/**
 * @param {string[]} strs
 * @return {string}
 */
var longestCommonPrefix = function (strs) {
    let firstWord = strs[0];
    let count = 0;
    let res = ''
    for (let count = 0; count < firstWord.length; count += 1) {
        let temp = res+firstWord[count]
        if (strs.every(s => s.startsWith(temp))) {
            res += firstWord[count]
        }

    }
    return res;
};

File: 0014-longest-common-prefix/0014-longest-common-prefix.rs
impl Solution {
    pub fn longest_common_prefix(strs: Vec<String>) -> String {
        let first_word = &strs[0];
        let remaining = &strs[1..];
        let mut res = String::new();
        for (i,ch) in first_word.chars().enumerate(){
             for word in remaining{
                    if word.chars().nth(i)  != Some(ch) {
                       return res;
                    }    
             }
             res.push(ch);
        }
        res
    }
}